docs-new/.github/workflows/ci.yml

```
@@ -0,0 +1,29 @@
+name: ci 
+on:
+  push:
+    branches:
+      - master 
+      - main
+permissions:
+  contents: write
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          git fetch --tags
+          git worktree add home/docs/1.3.1 docs-1.3.1
+          git worktree add home/javadoc javadoc
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+      - run: echo "cache_id=$(date --utc '+%V')" >> $GITHUB_ENV 
+      - uses: actions/cache@v3
+        with:
+          key: mkdocs-material-${{ env.cache_id }}
+          path: .cache
+          restore-keys: |
+            mkdocs-material-
+      - run: pip install -r requirements.txt
+      - run: mkdocs gh-deploy --force
```

This will overwrite the PyIceberg site 🕺 

I can prioritize the migration to a separate repo. I think we should get these in and then we're good to go: https://github.com/apache/iceberg/pull/8622 https://github.com/apache/iceberg/pull/8652 https://github.com/apache/iceberg/pull/8651

---

Just a heads-up for PyIceberg I had to jump through some hoops because it also requires a `CNAME` file: https://github.com/apache/iceberg/blob/master/.github/workflows/python-ci-docs.yml#L46-L56 Probably we also have to make sure that we deprecate the asf-site branch that currently hosts the docs.

---

In a follow up PR I think it is nice to add the `strict` flag to avoid slipping in errors.
```suggestion
      - run: mkdocs gh-deploy --force --strict
```

---

Totally agree, I think this won't be able to be applied in this initial PR though as I'm methodically trying to split up the work and this option would increase the scope of Phase 1: [See my comment](https://github.com/apache/iceberg/pull/8659#issuecomment-1737155953).

---

> I can prioritize the migration to a separate repo.

I thought we were going to try and build this on separate ASF infra or something. Either way this would be good to get done, but I wouldn't prioritize it over the Arrow work. I'll need that for the `Getting Started` pages ;). Which I care about much more.

---

> Totally agree, I think this won't be able to be applied in this initial PR though as I'm methodically trying to split up the work and this option would increase the scope of Phase 1: https://github.com/apache/iceberg/pull/8659#issuecomment-1737155953.

Yes, I wasn't suggesting to do it in this PR, but I just wanted to plant the seed.

> I thought we were going to try and build this on separate ASF infra or something. Either way this would be good to get done, but I wouldn't prioritize it over the Arrow work. I'll need that for the Getting Started pages ;). Which I care about much more.

With the current CI that you have in this repository, it will [overwrite the `gh-pages` branch](https://github.com/apache/iceberg/tree/gh-pages), and therefore whipping the PyIceberg docs. We can also deploy to the [ASF infra as we do with iceberg-docs](https://github.com/apache/iceberg-docs/tree/asf-site), but that requires something different than the `mkdocs gh-deploy` that's in there now. For reference: [docs on gh-deploy](https://www.mkdocs.org/user-guide/deploying-your-docs/#project-pages).

---

Ah got it, [I've ripped it out temporarily](https://github.com/apache/iceberg/blob/000344ebe85e26e8a97475494adbb5e62103d929/docs-new/.gitignore#L2) for this PR.